src/ngraph/op/function_call.cpp

```diff
@@ -51,7 +52,9 @@ op::FunctionCall::FunctionCall(shared_ptr<Function> function, const NodeVector&
 
 shared_ptr<Node> op::FunctionCall::copy_with_new_args(const NodeVector& new_args) const
 {
-    return make_shared<FunctionCall>(m_function, new_args);
+    shared_ptr<FunctionCall> fc = make_shared<FunctionCall>(m_function, new_args);
+    fc->m_function = clone_function(*m_function);
```

Comment:
I don't think we should simply clone the function here, in case a single function has multiple call sites. (Of course, it's hard to say how big a deal this actually is, since currently only the TF bridge is using embedded functions, and then only as a last resort.) I guess we would have to maintain a `FunctionMap` akin to the `NodeMap` over in `clone_function`, and pass `new_functions` in to `copy_with_new_args`.

Reply:
When we `compile()` a graph we mutate it. Running the same function through a CPU backend and then an NNP backend would cause mayhem, and probably fail, if we don't actually copy stuff.
copy_with_new_args should actually make a copy, it was not before this change as it maintained a pointer to the same function as the original op.
I don't understand the need for FunctionMap, please `splain

Reply:
I still think we should clone a function only once per graph-copy, i.e., if a function is used by more than one op in the original graph, then we should preserve that sharing in the new graph, albeit with a new copy of the function. I understand that it doesn't change anything semantically, but if someone is generating a graph that calls the same function in many places, this may be unexpected and undesired behavior.

That said, AFAIK we don't actually use multiple call sites in practice (in AFAIK TF is the only backend using embedded functions at all, and even there we try to avoid them whenever we can), so if we just document that functions with multiple call sites may be duplicated, I don't suppose it's that big a deal.